src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -766,151 +367,53 @@ private class PowerShellResult
-                    {
-                        string diagnosticId = GetUniqueIdFromDiagnostic(markerDiagnostic);
-                        fileCorrections[diagnosticId] = marker.Correction;
-                    }
+            fileCorrections.Corrections.Clear();
 
-                    diagnostics.Add(markerDiagnostic);
-                }
-            }
-            finally
+            for (int i = 0; i < markers.Count; i++)
             {
-                fileLock.Release();
-            }
+                ScriptFileMarker marker = markers[i];
 
-            // Always send syntax and semantic errors.  We want to
-            // make sure no out-of-date markers are being displayed.
-            _languageServer.Document.PublishDiagnostics(new PublishDiagnosticsParams()
-            {
-                Uri = new Uri(scriptFile.DocumentUri),
-                Diagnostics = new Container<Diagnostic>(diagnostics),
-            });
-        }
+                Diagnostic diagnostic = GetDiagnosticFromMarker(marker);
 
-        public async Task<IReadOnlyDictionary<string, MarkerCorrection>> GetMostRecentCodeActionsForFileAsync(string documentUri)
-        {
-            if (!_mostRecentCorrectionsByFile.TryGetValue(documentUri, out (SemaphoreSlim fileLock, Dictionary<string, MarkerCorrection> corrections) fileCorrectionsEntry))
-            {
-                return null;
-            }
-
-            await fileCorrectionsEntry.fileLock.WaitAsync().ConfigureAwait(false);
-            // We must copy the dictionary for thread safety
-            var corrections = new Dictionary<string, MarkerCorrection>(fileCorrectionsEntry.corrections.Count);
-            try
-            {
-                foreach (KeyValuePair<string, MarkerCorrection> correction in fileCorrectionsEntry.corrections)
+                if (marker.Correction != null)
                 {
-                    corrections.Add(correction.Key, correction.Value);
+                    string diagnosticId = GetUniqueIdFromDiagnostic(diagnostic);
+                    fileCorrections.Corrections[diagnosticId] = marker.Correction;
                 }
 
-                return corrections;
+                diagnostics[i] = diagnostic;
             }
-            finally
+
+            _languageServer.Document.PublishDiagnostics(new PublishDiagnosticsParams
             {
-                fileCorrectionsEntry.fileLock.Release();
-            }
+                Uri = new Uri(scriptFile.DocumentUri),
+                Diagnostics = new Container<Diagnostic>(diagnostics)
+            });
         }
 
-        // Generate a unique id that is used as a key to look up the associated code action (code fix) when
-        // we receive and process the textDocument/codeAction message.
-        internal static string GetUniqueIdFromDiagnostic(Diagnostic diagnostic)
+        private static ConcurrentDictionary<string, MarkerCorrection> CreateFileCorrectionsEntry(string fileUri)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused private method 'CreateFileCorrectionsEntry'.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4961943)